tests/integration/records/single_db/test_records_unload_datetime.py

```diff
@@ -0,0 +1,475 @@
+        }
+        for datetimeformattz in DATETIMETZ_CASES:
+            addl_hints: PartialRecordsHints = {}
+            pandas_compatible_addl_hints: PartialRecordsHints = {
+                'dateformat': matching_dateformat[datetimeformattz],
+                'datetimeformat': datetimeformattz.replace('OF', ''),
+            }
+            if self.engine.name == 'redshift':
+                if datetimeformattz != 'YYYY-MM-DD HH:MI:SSOF':
+                    # this is the only format supported by Redshift on
+                    # export, so we're going to need to be sure to use
+                    # hints that work in Pandas - i.e.,
+                    addl_hints = pandas_compatible_addl_hints
+                    if 'AM' in datetimeformattz:
+                        # TODO: Add a GitHub issue for this
+                        logger.warning('Cannot export this dateformattz using Pandas or Redshift--'
+                                       'skipping test')
+                        continue
+            elif self.engine.name == 'mysql':
+                # mysql has no exporter defined, so everything is via Pandas
+                addl_hints = pandas_compatible_addl_hints
+                if 'AM' in datetimeformattz:
+                    # TODO: Add a GitHub issue for this
+                    logger.warning('Cannot export this dateformat using Pandas, '
+                                   'which is needed on MySQL--skipping test')
+                    continue
+            elif self.engine.name == 'postgresql':
+                if datetimeformattz in ['YYYY-MM-DD HH:MI:SSOF',
+                                        'YYYY-MM-DD HH24:MI:SSOF']:
+                    # ensure we keep other areas of ISO style
+                    addl_hints = {
+                        'timeonlyformat': 'HH:MI:SS',
+                        'dateformat': 'YYYY-MM-DD',
+                        'datetimeformat': 'YYYY-MM-DD HH:MI:SS',
+                    }
+                else:
+                    # We will be using pandas
+                    addl_hints = pandas_compatible_addl_hints
+                    if 'AM' in datetimeformattz:
+                        # TODO: Add a GitHub issue for this
+                        logger.warning('Cannot export this dateformat using Pandas or PostgreSQL--'
+                                       'skipping test')
+                        continue
+            elif self.engine.name == 'vertica':
+                # Make sure our '\n' strings below are valid when comparing output
+                addl_hints['record-terminator'] = '\n'
+                if datetimeformattz not in ['YYYY-MM-DD HH:MI:SSOF',
+                                            'YYYY-MM-DD HH24:MI:SSOF']:
+                    # We will be using pandas
+                    addl_hints.update(pandas_compatible_addl_hints)
+                    if 'AM' in datetimeformattz:
+                        # TODO: Add a GitHub issue for this
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker